src/App.jsx

```diff
@@ -65,100 +53,107 @@ const styles = theme => ({
     padding: theme.spacing.unit * 3,
     minWidth: 0, // So the Typography noWrap works
   },
+  formControlLabel: {
+    display: 'flex',
+    align: 'center',
+    marginLeft: 0,
+  },
   toolbar: theme.mixins.toolbar,
 });
-
 const bugType = ['Unassigned Bug', 'Assigned Bugs', 'Simple Bugs'];
 
 class App extends Component {
-
   constructor(props) {
     super(props);
-    var allProjects = [...data.Projs];
-    allProjects.forEach(function(project, index) {
-      project.push(false)
-    })
+
+    const isProjectSelected = {};
+
+    Object.entries(projectGroups).forEach(projectGroup => {
+      projectGroup[1].forEach(
+        project => (isProjectSelected[project.fileName] = false)
+      );
+    });
+
     this.state = {
       mobileOpen: false,
-      allProjects : [...data.Projs],
+      projectGroups,
+      isProjectSelected,
     };
   }
 
-
   handleDrawerToggle = () => {
     this.setState({ mobileOpen: !this.state.mobileOpen });
   };
 
-  handleCheckboxToggle = (proj) => {
-    var allProjects = [...this.state.allProjects];
+  handleCheckboxToggle = key => {
+    const { isProjectSelected } = this.state;
 
-    allProjects.forEach(function(project, index) {
-      if (project[0] === proj) {
-        console.log(project[2])
-        project[2] = !project[2];
-      }
-    });
-
-    this.setState({allProjects: allProjects});
+    isProjectSelected[key] = !isProjectSelected[key];
+    this.setState({ isProjectSelected });
   };
 
   render() {
     const { classes, theme } = this.props;
-    const { projectSelected, allProjects } = this.state;
-
+    const { projectGroups, isProjectSelected } = this.state;
     const drawer = (
       <div>
         <div className={classes.toolbar} />
-          <FormControl component="fieldset">
-            <FormLabel component="legend">Are you interested in:</FormLabel>
-            <FormGroup>
-              { allProjects.map((proj, index) => (              
-                <FormControlLabel
-                  control={
-                    <Checkbox
-                      value={proj[0]}
-                      checked = {proj[2]}
-                      onChange = {(e) => this.handleCheckboxToggle(proj[0],e)}
-                    />
-                  }
-                  key={index}
-                  label={proj[0]}
-                />
-              ))}
-            </FormGroup>
+        <FormLabel component="legend">Are you interested in:</FormLabel>
 
-            <FormLabel component="legend">Do you know ?</FormLabel>
-            <FormGroup>
-              { data.Languages.map((lang, index) => (              
+        <FormGroup>
+          {Object.entries(projectGroups).map(([group, projects]) => (
+            <div key={group}>
+              <Typography variant="subheading">{group}</Typography>
```

Comment:
Can we make use of the `List` component here? This would add styling for list items like paddings and we would be able to add the on hover effect to when a user hovers over the items.

```js
{Object.entries(projectGroups).map(([group, projects]) => (
  <List>
    <ListItem>
      <ListItemText>{group}</ListItemText>
    </ListItem>
    {projects.map(project => (
      <ListItem key={`${group}-${project.fileName}`} id={project.fileName} onClick={this.handleCheckboxToggle} button>
        <Checkbox
          value={project.fileName}
          checked={isProjectSelected[project.fileName]}
        />
        <ListItemText primary={project.name} />
      </ListItem>
    ))}
  </List>
))}
```

Notice that I also changed

```js
onChange={e =>this.handleCheckboxToggle(project.fileName, e)}
```
to:

```js
id={project.fileName} onClick={this.handleCheckboxToggle}
```

Because of that, you will need to change `handleCheckboxToggle` to extract the value from the id:

```js
handleCheckboxToggle = ({ currentTarget: { id } }) => {
  const { isProjectSelected } = this.state;

  isProjectSelected[id] = !isProjectSelected[id];
  this.setState({ isProjectSelected });
};
```